.github/workflows/continuous-integration.yml

```diff
@@ -359,6 +359,111 @@ jobs:
           name: cypress-screenshot-artifacts
           path: cypress/screenshots
 
+      - name: Archive Cypress Mochawesome test results
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: cypress-mochawesome-test-results-artifact-${{ matrix.ci_node_index }}
+          path: cypress/results
+
+  mochawesome-report:
+    name: Generate Mochawesome Report
+    runs-on: ubuntu-latest
+    needs: cypress-tests
+    if: ${{ always() && (needs.cypress-tests.result == 'success' || needs.cypress-tests.result == 'failure') }}
+
+    defaults:
+      run:
+        working-directory: testing-tools-team-dashboard-data
+
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: "us-gov-west-1"
+
+      - name: Get BigQuery service credentials
+        uses: marvinpinto/action-inject-ssm-secrets@latest
+        with:
+          ssm_parameter: /dsva-vagov/testing-team/bigquery_service_credentials
+          env_variable_name: BIGQUERY_SERVICE_CREDENTIALS
+
+      - name: Get GitHub token
+        uses: marvinpinto/action-inject-ssm-secrets@latest
+        with:
+          ssm_parameter: /devops/VA_VSP_BOT_GITHUB_TOKEN
+          env_variable_name: VA_VSP_BOT_GITHUB_TOKEN
+
+      - name: Checkout Testing Tools Team Dashboard Data repo
+        uses: actions/checkout@v2
+        with:
+          repository: department-of-veterans-affairs/testing-tools-team-dashboard-data
+          token: ${{ env.VA_VSP_BOT_GITHUB_TOKEN }}
+          path: testing-tools-team-dashboard-data
+
+      - name: Get Node version
+        id: get-node-version
+        run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
+
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ steps.get-node-version.outputs.NODE_VERSION }}
+
+      - name: Setup Cloud SDK
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: vsp-analytics-and-insights
+          service_account_key: ${{ env.BIGQUERY_SERVICE_CREDENTIALS }}
+          export_default_credentials: true
+
+      - name: Download test result artifact 0
+        uses: actions/download-artifact@v2
+        with:
+          name: cypress-mochawesome-test-results-artifact-0
+          path: testing-tools-team-dashboard-data/src/mochawesome/data
+        continue-on-error: true
```

Comment:
@holdenhinkle: I added this setting in case any of the artifacts didn't successfully get saved and/or downloaded. This means that even if one of the test results are missing, the rest of the results will get reported. Hopefully this still aligns with the behavior you expected, but if not, I can change it back.

Reply:
I think this would be a rare occurrence in any case. So far, it's only happened when one batch of tests didn't even start because Cypress couldn't access the port (but that's a separate issue to resolve).

Reply:
Hey Eugene,

I noticed the `mochawesome-report` job failed yesterday because one of the 6 instances running the Cypress tests had an error because that randomly selected port was already in use (or something like that).

If results are reported when tests results are missing, our numbers will get thrown off. We need all results, or no results. Here's why:

Example of some of the data that's stored in BigQuery for each Cypress test execution (it's created from merging the Mochawesome results):

```
[
  {
    "suites": "155",
    "tests": "309",
    "passes": "281",
    "pending": "26",
    "failures": "0",
    "start": "2021-06-21T20:09:54.583Z",
    "end": "2021-06-21T20:27:40.529Z",
    "duration": "2066266",
    "testsRegistered": "309",
    "passPercent": "90.93851132686085",
    "pendingPercent": "8.414239482200648",
    "other": "0",
    "hasOther": false,
    "skipped": "2",
    "hasSkipped": true,
    "id": "232ebf96-589e-409b-a329-8e0079e055c8",
    "mochawesomeReportUrl": "https://reimagined-telegram-c11df05a.pages.github.io/cypress-reports/232ebf96-589e-409b-a329-8e0079e055c8.html"
  }
]
```

We track values like `suites` (number of spec files) and `tests`. These number will get thrown off if all of the tests don't run.

It would be great if you can change it back. Thanks Eugene.